include/pinocchio/multibody/joint/joint-ellipsoid.hpp

```diff
@@ -0,0 +1,555 @@
+  {
+    EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+    typedef JointEllipsoidTpl<_Scalar, _Options> JointDerived;
+    PINOCCHIO_JOINT_DATA_TYPEDEF_TEMPLATE(JointDerived);
+    PINOCCHIO_JOINT_DATA_BASE_DEFAULT_ACCESSOR
+
+    ConfigVector_t joint_q;
+    TangentVector_t joint_v;
+
+    Constraint_t S;
+    Transformation_t M;
+    Motion_t v;
+    Bias_t c;
+
+    // [ABA] specific data
+    U_t U;
+    D_t Dinv;
+    UD_t UDinv;
+    D_t StU;
+
+    JointDataEllipsoidTpl()
+    : joint_q(ConfigVector_t::Zero())
+    , joint_v(TangentVector_t::Zero())
+    , S()
+    , M(Transformation_t::Identity())
+    , v(Motion_t::Zero())
+    , c(Bias_t::Zero())
+    , U(U_t::Zero())
+    , Dinv(D_t::Zero())
+    , UDinv(UD_t::Zero())
+    , StU(D_t::Zero())
+    {
+    }
+
+    static std::string classname()
+    {
+      return std::string("JointDataEllipsoid");
+    }
+    std::string shortname() const
+    {
+      return classname();
+    }
+
+  }; // struct JointDataEllipsoidTpl
+
+  PINOCCHIO_JOINT_CAST_TYPE_SPECIALIZATION(JointModelEllipsoidTpl);
+  template<typename _Scalar, int _Options>
+  struct JointModelEllipsoidTpl : public JointModelBase<JointModelEllipsoidTpl<_Scalar, _Options>>
+  {
+    EIGEN_MAKE_ALIGNED_OPERATOR_NEW
+    typedef JointEllipsoidTpl<_Scalar, _Options> JointDerived;
+    PINOCCHIO_JOINT_TYPEDEF_TEMPLATE(JointDerived);
+
+    typedef JointModelBase<JointModelEllipsoidTpl> Base;
+    using Base::id;
+    using Base::idx_q;
+    using Base::idx_v;
+    using Base::idx_vExtended;
+    using Base::setIndexes;
+
+    typedef typename Transformation_t::Vector3 Vector3;
+
+    Scalar radius_a;
+    Scalar radius_b;
+    Scalar radius_c;
+
+    JointDataDerived createData() const
+    {
+      return JointDataDerived();
+    }
+
+    /// @brief Default constructor. Creates a sphere (0.01, 0.01, 0.01).
+    JointModelEllipsoidTpl()
+    : radius_a(Scalar(0.01))
+    , radius_b(Scalar(0.01))
+    , radius_c(Scalar(0.01))
+    {
+    }
+
+    /// @brief Constructor with specified radii.
+    /// @param a Semi-axis length along x-direction
+    /// @param b Semi-axis length along y-direction
+    /// @param c Semi-axis length along z-direction
+    JointModelEllipsoidTpl(const Scalar & a, const Scalar & b, const Scalar & c)
+    : radius_a(a)
+    , radius_b(b)
+    , radius_c(c)
+    {
+    }
+
+    const std::vector<bool> hasConfigurationLimit() const
+    {
+      return {true, true, true};
+    }
+
+    const std::vector<bool> hasConfigurationLimitInTangent() const
+    {
+      return {true, true, true};
+    }
+
+    /// @brief Computes the spatial transformation M(q) from joint configuration.
+    /// @param[in] c0, s0 Cosine and sine of q[0]
+    /// @param[in] c1, s1 Cosine and sine of q[1]
+    /// @param[in] c2, s2 Cosine and sine of q[2]
+    /// @param[out] data Joint data where M will be stored
+    void computeSpatialTransform(
+      Scalar c0, Scalar s0, Scalar c1, Scalar s1, Scalar c2, Scalar s2, JointDataDerived & data)
```

Comment:
```suggestion
      const Scalar& c0, const Scalar& s0, const Scalar& c1, const Scalar& s1, const Scalar& c2, const Scalar& s2, JointDataDerived & data) const
```

This will avoid a copy when using scalar like casadi/cppad/cppadcg/boostmpfr